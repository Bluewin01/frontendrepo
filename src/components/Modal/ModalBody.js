import React from "react";
import { Modal, Form } from "react-bootstrap";
import "./_ModalStyle.css";

export const ModalBody = (props) => {
  const { content } = props;
  return (
    <Modal.Body>
      <Form>{content}</Form>
    </Modal.Body>
  );
};
