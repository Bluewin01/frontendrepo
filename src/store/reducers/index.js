import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
