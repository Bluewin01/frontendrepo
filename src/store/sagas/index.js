import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
