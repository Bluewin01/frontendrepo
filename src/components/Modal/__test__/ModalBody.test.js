import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../../store";
import { Provider } from "react-redux";
import { ModalBody } from "../ModalBody";

describe("Component: Modal", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  afterEach(cleanup);
  it("should be render modalBody correctly", () => {
    const mockUseState = () => jest.fn();
    const inputfields = [
      {
        setInput: mockUseState,
        value: "to",
        showError: "showErrorMessageTo",
        name: "To",
        type: "text",
      },
    ];

    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <ModalBody inputfields={inputfields} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
