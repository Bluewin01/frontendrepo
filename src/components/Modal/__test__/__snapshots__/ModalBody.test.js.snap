// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Modal should be render modalBody correctly 1`] = `
<div
  className="modal-body"
>
  <form
    className=""
  />
</div>
`;
